processor/sourceprocessor/source_processor.go

```diff
@@ -41,27 +64,132 @@ type attributeFiller struct {
 	labels          []string
 }
 
+type sourceTraceProcessor struct {
+	collector             string
+	source                string
+	sourceCategoryFiller  attributeFiller
+	sourceNameFiller      attributeFiller
+	sourceHostFiller      attributeFiller
+	excludeNamespaceRegex *regexp.Regexp
+	excludePodRegex       *regexp.Regexp
+	excludeContainerRegex *regexp.Regexp
+	excludeHostRegex      *regexp.Regexp
+	nextConsumer          consumer.TraceConsumer
+	keys                  sourceTraceKeys
+}
+
 const (
 	alphanums = "bcdfghjklmnpqrstvwxz2456789"
 
-	annotationPrefix                = "pod_annotation_"
-	podTemplateHashKey              = "pod_labels_pod-template-hash"
-	podNameKey                      = "pod_name"
-	sourceHostSpecialAnnotation     = annotationPrefix + "sumologic.com/sourceHost"
-	sourceNameSpecialAnnotation     = annotationPrefix + "sumologic.com/sourceName"
-	sourceCategorySpecialAnnotation = annotationPrefix + "sumologic.com/sourceCategory"
+	sourceHostSpecialAnnotation     = "sumologic.com/sourceHost"
+	sourceNameSpecialAnnotation     = "sumologic.com/sourceName"
+	sourceCategorySpecialAnnotation = "sumologic.com/sourceCategory"
+
+	includeAnnotation = "sumologic.com/include"
+	excludeAnnotation = "sumologic.com/exclude"
 )
 
+func compileRegex(regex string) *regexp.Regexp {
+	if regex == "" {
+		return nil
+	}
+
+	re, err := regexp.Compile(regex)
+	if err != nil {
+		log.Fatalf("Cannot compile reguler expression: %s Error: %v\n", regex, err)
+	}
+
+	return re
```

Comment:
What about returning nil regex when error happens? Will `regexp.Compile(regex)` return nil in the first element of the tuple?

Reply:
I believe that in such case `log.Fatalf` will cause the application to exit, let me check that

Reply:
Yeap:
```
2020/05/15 14:04:25 Cannot compile reguler expression: [ Error: error parsing regexp: missing closing ]: `[`

Process finished with exit code 1
```

Reply:
(Just a small typo to fix in message) :)